### DISCUSSION TITLE
Consider no breaking changes on minor version update

### DISCUSSION THREAD

**Discussion by cbeauchesne** [python-jsonschema/jsonschema#1005] [Q&A] [answered]
Hi team !

About this change in [v4.16.0](https://github.com/python-jsonschema/jsonschema/blob/main/CHANGELOG.rst#v4160) : 

> Accessing jsonschema.draftN_format_checker is deprecated. Instead, if you want access to the format checker itself, it is exposed as jsonschema.validators.DraftNValidator.FORMAT_CHECKER on any jsonschema.protocols.Validator

If for some reason, we need to support several version of a third parties (for instance, for packages widely used where fixing a version for third parties is a very bad practice => incompabilities between two packages), the usual practice is to support any 4.x version, as we expect no breaking change for all 4.x versions.

Could it be possible to consider semver principles, meaning no breaking change on minor version update ?

Thank you for your great work though, I use jsonschema for all my schema validation, it does a great job! 🙇

**Comment by Julian** [comment, marked as answer]
Thanks for the kind words. Yes, the removal will/would happen alongside a major version bump!
